api/src/libs/githubInvitationInit.ts

```diff
@@ -0,0 +1,188 @@
+import { createAppAuth } from '@octokit/auth'
+import { request } from '@octokit/request'
+import { every, isArray, isObject, isString, intersectionBy} from 'lodash'
+import { getConfig, getGithubPrivateKey } from '../config/githubConfig'
+import { logger } from '@bcgov/common-nodejs-utils';
+
+// cached value
+type APPInitailValue = {
+  initialized: number | null,
+  apps: object,
+  nonInstallatedApp: Function | null,
+}
+
+const  installationApps = <APPInitailValue>{
+  initialized: null,
+  apps: {},
+  nonInstallatedApp: null
+}
+
+/**
+ * getNonInstallationApp
+ * @returns a non installed github app
+ */
+export const getNonInstallationApp = () => {
+    logger.info('getNonInstallationApp')
+  // caches a non installed app
+  try{
+
+
+  if (!installationApps.nonInstallatedApp) {
+    const auth = createAppAuth({
+      appId: 128566,
+      privateKey: getGithubPrivateKey(),
+      clientId: process.env.CLIENT_ID,
+      clientSecret: process.env.CLIENT_SECRET,
+    })
+
+    installationApps.nonInstallatedApp = request.defaults({
+      request: {
+        hook: auth.hook,
+      },
+      mediaType: {
+        previews: ['machine-man'],
+      },
+    })
+  }
+ } catch(e){
+    logger.error('I catch an error', e);
+    
+
+  }
+
+
+  return installationApps.nonInstallatedApp
+}
+
+const newAuthorizedApp = (installationId) => {
+  const app = createAppAuth({
+    appId: 128566,
+    privateKey: getGithubPrivateKey(),
+    clientId: process.env.CLIENT_ID,
+    clientSecret: process.env.CLIENT_SECRET,
+    installationId,
+  })
+
+  return {
+    initialized: Date.now(),
+    app,
+    id: installationId,
+    authenticatedRequest: request.defaults({
+      request: {
+        hook: app.hook,
+      },
+      mediaType: {
+        previews: ['machine-man'],
+      },
+    }),
+  }
+}
+
+export const getInstallations = async () => {
+    logger.info('getInstallations')
+
+    const nonInstallationRequest = getNonInstallationApp()
+   
+    
+    if(nonInstallationRequest !== null){
+      const response = await nonInstallationRequest('GET /app/installations')
+      return response.data
+    }
+    //TODO: maybe throw if there's no installation
+    return []
+    // console.log('hiahia what is response',response.data)
+
+}
+
+export const getOrgInstallations = async () => {
+  logger.info('getOrgInstallations')
+  const config = getConfig()
+
+  if (
+    !config ||
+    !isObject(config) ||
+    !isArray(config.orgs) ||
+    !every(config.orgs, isString) ||
+    config.orgs.length === 0
+  ) {
+    throw new Error(
+      'Configuration is invalid. config.orgs is invalid or misconfigured'
+    )
+  }
```

Comment:
you dont need to validate the github apps installations since this is only being installed bcgov/bcdevops